tests/test_lichess_bot.py

```diff
@@ -97,11 +97,10 @@ def test_decline_challenge(self):
         )
 
         assert challenge_event
-        accepted = sporkfish._event_action_accept_challenge(challenge_event)
-        assert not accepted
+        assert not sporkfish._event_action_accept_challenge(challenge_event)
         try:
             sporkfish.client.bots.abort_game(challenge_event["challenge"]["id"])
-            assert False, "Expected to fail to abort game, as challenge was declined"
+            assert False, "Expected to fail to abort game as challenge was declined."
         except RetryError:
             # This is a success
             pass
```

Comment:
would you not write a test to get a bot to resign the game and see if sporkfish exits the play_move function?
```
for state in states:
            logging.debug(f"Game state: {state}")
            if state["type"] == "gameState":
                self._play_move(color, state["moves"], game_id, state)
            elif state["type"] == "gameStateResign":
                return "Game concluded with resignation from opponent"
```

Reply:
Good shout for the test. I test it locally and got lazy. 

Not so keen on returning the sentence because it doesn't do anything atm. If we do decide to return something I'd suggest returning a termination code/enum. String comparison is not as good a practice as enum type comparison.

Reply:
Edit: getting if the resul was done in multiprocess is painful, I'll add the termination enum.

Reply:
I struggled alot to get the right situation to occur (i.e. resign must be in the for loop of handlin states that are not gameFull) so I changed up the design a little and allow mocking here.